plugins/events_history_plugin/src/events_history/mgr/events_history_mgr.py

```diff
@@ -0,0 +1,125 @@
+#
+# Copyright © 2013-2022 NVIDIA CORPORATION & AFFILIATES. ALL RIGHTS RESERVED.
+#
+# This software product is a proprietary product of Nvidia Corporation and its affiliates
+# (the "Company") and all right, title, and interest in and to the software
+# product, including all associated intellectual property rights, are and
+# shall remain exclusively with the Company.
+#
+# This software product is governed by the End User License Agreement
+# provided with the software product.
+# @author: Abeer Moghrabi
+# @date:   Aug 9, 2023
+#
+
+
+from utils.singleton import Singleton
+import fnmatch
+import os
+import gzip
+from utils.logger import Logger, LOG_LEVELS
+from model.event import Event
+from constants.events_constants import EventsConstants
+import re
+
+
+class EventsHistoryMgr(Singleton):
+    """
+    This class is designed to manage UFM events history
+    """
+
+    def __init__(self):
+
+        self._events = EventsCollection(Event)
+        self.parse_event_log_files()
+
+    events = property(lambda self: self._events)
+
+    def openEventFile(self, fname, mode):
+        """
+        Opens an event log file whether it is a compressed file or a regular one
+        @param fname: the event log file name to be opened
+        @param mode: a string indicating how the file is to be opened.
+        @return: a reference to the opened file
+        """
+        if fname.endswith(".gz"):
+            return gzip.open(fname, mode, encoding='utf8')
+        else:
+            return open(fname, mode)
```

Comment:
from where you imported this function? it seems you forget the import statement

Reply:
The open() function is a built-in function that is part of the core Python language, so you do not need to import it